src/super_gradients/training/utils/media/image.py

```diff
@@ -77,12 +83,17 @@ def load_image(image: ImageSource) -> np.ndarray:
         - str:              A string representing either a local file path or a URL to an image
 
     :param image: Single image of supported types.
-    :return:      Image as numpy arrays. If loaded from string, the image will be returned as RGB.
+    :return:      Image as numpy arrays (H, W, C). If loaded from string, the image will be returned as RGB.
     """
     if isinstance(image, np.ndarray):
+        if image.shape[0] == 3:
```

Comment:
Hard-coded number of input channels of 3 should go away IMO. Instead we should _somehow_ get the number of input channels from the model itself.

Reply:
`load_image` expects a single image. In the `predict`, we use `load_images` which itself calls `load_image` per image.

We could add a check to make sure a single image is passed.
Or, we could instead move this `if image.shape[0] == 3:` logic into `load_images` and that way "optimize" the code, by not doing it per image, and not "spliting" the batch tensor into multiple single-image array.
I'll consider the second option and see if it all works easily